File: offer/15.二进制中1的个数.js
function hammingWeight (n) {
  let res = 0;
  while (n) {
    res++;
    n &= (n - 1);
  }
  return res;
};

function hammingWeight(n) {
  let res = 0;
  while (n !== 0) {
    res += n % 2;
    n = Math.floor(n / 2);
  }
  return res;
}

File: offer/15.二进制中1的个数.go
func hammingWeight(n uint32) int {
	res := 0
	for n != 0 {
		res += int(n % 2)
		n /= 2
	}
	return res
}

func hammingWeight(n uint32) int {
	res := 0
	for n != 0 {
		res++
		n &= (n - 1)
	}
	return res
}